src/components/forms/change-password-form/change-password-form.js

```diff
@@ -40,10 +40,15 @@ const validate = ({ oldPassword, newPassword, confirmPassword }) => {
 };
 
 const ChangePasswordReduxForm = (props) => {
-  const { submitting, onSubmit, handleSubmit, error } = props;
+  const { submitting, onSubmit, handleSubmit, error, reset } = props;
 
   return (
-    <Form onSubmit={handleSubmit(onSubmit)}>
+    <Form
+      onSubmit={handleSubmit((data) => {
+        reset();
+        return onSubmit(data);
```

Comment:
bind a function as submit instead.
why are tou returning the submit function?

Reply:
If I get it correctly...

Normally one would do:
```
onSubmit={handleSubmit(onSubmit)}
```

handleSubmit(onSubmit) returns a function that takes the event as an argument. It then sends the event to the onSubmit.
In other words handleSubmit(onSubmit)(event) can be written as:
```
handleSubmit(event => onSubmit(event))
```

which can be written as:
```
handleSubmit(event => {
  return onSubmit(event);
})
```

The code below is my way of doing the exact same thing but inject a execution of this.props.reset():

```
handleSubmit((event) => {
  reset();
  return onSubmit(event);
}
```